### DISCUSSION TITLE
Any direct association to DWMBlurGlass (like engine, etc)?

### DISCUSSION THREAD

**Discussion by SoftwareType** [ALTaleX531/OpenGlass#44] [Q&A] [answered]
Does OpenGlass use the same software engine-ish that DWMBlurGlass uses? Any other association?

**Comment by ALTaleX531** [comment, marked as answer]
This is a good question.

The dcomp branch of OpenGlass was developed in response to DWMBlurGlass's need for a custom blur method, except that it was still called [AcrylicEverywhere](https://github.com/ALTaleX531/AcrylicEverywhere/) at that time. DWMBlurGlass used version 1.0 of the glass engine when it was released in 2.0, and version 1.1 for releases 2.1 and later. However, today's OpenGlass is 1.2 after several updates, with complete occlusion optimization and effect input optimization. DWMBlurGlass hasn't been updated in a long time, so it hasn't migrated the latest glass engine to it.
As for the legacy branch of OpenGlass, it has nothing to do with DWMBlurGlass, I developed it simply because of the curiosity of the glass8 implementation and the fact that there are still people complaining about the terrible performance of the dcomp branch (which I don't think is true). dcomp branch uses `DirectComposition` and `Windows.UI.Composition` as a glass renderer, while the legacy branch is a glass renderer that I wrote myself using `Direct2D`.

As you know, I'm actually also the developer of DWMBlurGlass, although I'm not directly responsible for it. But I've planned so many exclusive features for it, like allowing users to write their own shaders using xml, custom window transition animations. But sadly, I don't think it will be long before I'm not active in the Windows modding community.

**Reply by Skyyblaze** [reply]
I'm very sorry to hear that, you and @Maplespe did amazing work for the Windows modding community and tools like DWMBlurGlass and OpenGlass. Is there any way to donate to either of you to show some appreciation?

**Reply by ALTaleX531** [reply]
Maplespe has placed his donation link on the DWMBlurGlass repository long ago, donate to him if you want.

**Reply by Skyyblaze** [reply]
Alright, I'll definitely donate something, you both are great people!

**Reply by TheInsane101** [reply]
Aww, it really sucks to hear that the custom shaders idea is cancelled. But it's great what we have already! Thank you so much for all you've done for us!

Can the custom shaders thing also be done open source or is it not possible without your implementation?
